src/openrct2/util/SawyerCoding.cpp

```diff
@@ -53,35 +51,34 @@ size_t SawyerCodingWriteChunkBuffer(uint8_t* dst_file, const uint8_t* buffer, Sa
             // fwrite(buffer, 1, chunkHeader.length, file);
             break;
         case CHUNK_ENCODING_RLE:
-            encode_buffer = static_cast<uint8_t*>(malloc(0x600000));
-            chunkHeader.length = static_cast<uint32_t>(EncodeChunkRLE(buffer, encode_buffer, chunkHeader.length));
-            std::memcpy(dst_file, &chunkHeader, sizeof(SawyerCodingChunkHeader));
-            dst_file += sizeof(SawyerCodingChunkHeader);
-            std::memcpy(dst_file, encode_buffer, chunkHeader.length);
-
-            free(encode_buffer);
+            {
+                std::vector<uint8_t> encode_buffer(0x600000);
```

Comment:
In such cases, it's easier to use `std::make_unique_for_overwrite`, especially since that will not default-initialize the entire buffer, unlike your vector approach.

Reply:
Thank you for your feedback!

It seems that `std::make_unique_for_overwrite` is only available since C++20, and this code is built with `-std=gnu++17` compiler flag, so unfortunately we cannot use it.
I could use regular `std::make_unique`, but that wouldn't help with potential performance hit (due to the default-init), compared to `std::make_unique_for_overwrite` or `malloc()`.

Any thoughts on that?

Reply:
It's not the first time I made this mistake - I feel like maybe it makes sense for the codebase to "backport" `make_unique_for_overwrite`.

Reply:
It may be, but that would most likely be out of scope for this PR.
So what would be your preferred solution for now?

Reply:
No strong preference one way or the other then, but I'd consider `make_unique` the best still.

Reply:
Thank you for promptly replay.
In that case I consider this part as resolved, since that is what I have already changed it to.
Feel free to reopen, should you have any other comments.